src/index.js

```diff
@@ -33,39 +47,80 @@ const configuration = {
       // You can still add or override specific transforms if needed.
       // For example, if you wanted a different naming convention.
       transforms: ['name/kebab'],
-      files: [
-        {
-          format: 'css/variables',
-          destination: 'variables.css',
-          // Filter out tokens that are specific to Figma and not needed in the final CSS.
-          filter: (token) => !token.path.includes('figma'),
-          options: {
-            // This ensures that your CSS variables can reference other CSS variables.
-            outputReferences: true,
-            // Example of other options you could use:
-            // selector: '.gdb', // to scope variables to a specific class
-            // outputReferenceFallbacks: true // to add fallbacks for references
-          },
-        },
-      ],
+      files: [],
     },
   },
 };
 
-// Create a new Style Dictionary instance with the configuration.
-const sd = new StyleDictionary(configuration);
+/**
+ * @param {'Light'|'Dark'} mode
+ */
+function modifyTokensForMode(mode) {
+    // eslint-disable-next-line no-console
+    console.log(`########## Generating ${mode} mode tokens file ##########`);
+    const tokens = JSON.parse(fs.readFileSync('tokens/tokens.json', 'utf8'));
+
+    const inverseMode = mode === 'Light' ? 'Dark' : 'Light';
+    const theme = `theme/${inverseMode.toLowerCase()}`;
+
+    if(tokens[theme]) {
+      delete tokens[theme];
+    }
+
+    const themeTokensIndex = tokens['$themes'].findIndex(t => t.name === inverseMode);
+    if(themeTokensIndex === -1) {
+      throw new Error(`No theme with name ${inverseMode} found in $themes`);
+    }
+    tokens['$themes'].splice(themeTokensIndex, 1);
+
+    const tokensetIndex = tokens['$metadata']['tokenSetOrder'].findIndex((set) => set === theme);
+    tokens['$metadata']['tokenSetOrder'].splice(tokensetIndex, 1);
+
+    const modesDir = path.join(process.cwd(), 'modes');
+    fs.writeFileSync(path.join(modesDir, `${mode.toLowerCase()}-mode-tokens.json`), JSON.stringify(tokens, null, 2));
+}
 
 /**
  * Main build function to run Style Dictionary.
  * Using an async function allows us to use top-level await.
  */
 async function main() {
```

Comment:
Both output css would contain all common variable, which are the majority of them.

Have you considered to have one set (eg. Light) have the whole token set (without the Dark tokens) and then have the other only contains it's own with a `:root.dark` selector? When set it would override only the overlapping vars.

Reply:
I have, but I've no idea how to do it without further manipulation of the tokens.json TBH.